File: Semester5/MOBILE/Lab2/src/main/java/com/example/swimmingteamtracker/data/Swimmer.kt
package com.example.swimmingteamtracker.data

import android.os.Parcelable
import kotlinx.parcelize.Parcelize


@Parcelize
data class Swimmer(
    var fullName: String, var gender: String, var nationality: String, var weight: Double,
    var height: Int, var id: Int) : Parcelable {

        companion object {
            var currentId = 0;
        }

        constructor(
            fullName: String, gender: String, nationality: String, weight: Double,
            height: Int
        ) : this(fullName, gender, nationality, weight, height, currentId++) {
            }
    }

File: Semester5/MOBILE/Lab3/src/entities/Swimmer.ts
export default class Swimmer {
    private id : number;
    private fullname : string;
    private gender : string;
    private nationality : string;
    private weight : number;
    private height : number;
  
    constructor(id : number, fullname : string, gender : string, nationality : string, weight : number, height : number) {
      this.id = id;
      this.fullname = fullname;
      this.gender = gender;
      this.nationality = nationality;
      this.weight = weight;
      this.height = height;
    }
  
    getId() : number {
      return this.id;
    }
  
    getFullName() : string {
      return this.fullname;
    }
    
    getGender() : string {
      return this.gender;
    }
  
    getNationality() : string {
      return this.nationality;
    }
  
    getWeight() : number {
      return this.weight;
    }
  
    getHeight() : number {
      return this.height;
    }  

    resetId(newId: number) {
      this.id = newId;
    }
  }
    
    